vm/src/obj/objbyteinner.rs

```diff
@@ -25,6 +25,18 @@ pub struct PyByteInner {
     pub elements: Vec<u8>,
 }
 
+pub fn vec_from_str(value: &str, encoding: &str, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
+    let encoding = encoding.to_lowercase();
+    if encoding == "utf-8" || encoding == "u8" || encoding == "utf8" || encoding == "utf_8" {
```

Comment:
Thinking bytes.decode I drafted a normalize_encoding like cpython. maybe it could be usable :
```rust
//same algorithm as cpython
pub fn normalize_encoding(encoding: &str) -> String {
    let mut res = String::new();
    let mut punct = false;

    for c in encoding.chars() {
        if c.is_alphanumeric() || c == '.' {
            if punct && !res.is_empty() {
                res.push('_')
            }
            res.push(c.to_ascii_lowercase());
            punct = false;
        } else {
            punct = true;
        }
    }
    res
}
```
you'll see on the same file some ideas to deal with errors.
https://github.com/jgirardet/RustPython/blob/1736d63932d5b7b4a6093eed62019255545888b2/vm/src/stdlib/_codecs.rs#L66

Reply:
Both the function and the codec module looks really good. And I just recognized it is not a part of 847 😞 . Will you create a small PR with the part to adapt? Or do you prefer I import some required parts from your work? Just waiting for future and editing later will work too if it will be finally merged in future.

Reply:
I 'm waiting #847 to be merge, and next there is another big PR (depending of 847) with other bytes method. I 'll wait everything is reviewed/merge( it's much work) and then I will propose something for bytes.decode.
For no just take what you need, i'll see after.

Reply:
I added this function in `objbyteinner.rs`, thanks